src/reducers/semantics.js

```diff
@@ -0,0 +1,46 @@
+import config from "../app.config";
+import {
+	SEMANTIC_LOADING,
+	SEMANTIC_LOAD_COMPLETE,
+	LOGIN_ERROR,
+	RECEIVE_SEMANTICS,
+	DELETE_ITEM,
+	DELETE_ERROR,
+	RESET_ERROR
+} from "../actions";
+import {SemanticState, Semantics, Semantic} from "../utils/flowtype";
+
+type SemanticAction = {
+	type: RECEIVE_SEMANTICS,
+	semantics: Semantics,
+	item: Semantic,
+	loading: boolean
+}
+
+const defaultState = {semantics: [], loading: false};
+
+const semantics = (state:SemanticState = defaultState, action:SemanticAction) => {
+	switch(action.type) {
+	case RECEIVE_SEMANTICS:
+		return Object.assign({}, state, {semantics: action.semantics});
+	case DELETE_ITEM:
```

Comment:
there is no delete?

Reply:
I thought delete functionality was to be handled by us and not the users?

Reply:
Then you don't need to add to reducer? :-)